src/main/java/de/thomas_oster/liblasercut/drivers/GenericGcodeDriver.java

```diff
@@ -494,45 +500,73 @@ protected void setSpeed(double speedInPercent) {
   protected void setPower(double powerInPercent) {
     nextPower = powerInPercent/100.0*spindleMax;
   }
-
+  
   protected void setFocus(PrintStream out, double focus) throws IOException {
-    if (currentFocus != focus)
-    {
-      sendLine("G0 Z%f", focus);
-      currentFocus = focus;
+    Locale locale  = new Locale("en", "US");
+    String formatPattern = "###.##";
+    DecimalFormat coordinateFormat = (DecimalFormat)NumberFormat.getNumberInstance(locale);
+    coordinateFormat.applyPattern(formatPattern);
+    coordinateFormat.setMaximumFractionDigits(getGCodeDigits());
```

Comment:
Can you please deduplicate this code block? It is present about four times; a separate function would be easier to read. For example `String formatFloat(float value, int decimalPlaces)`.